internal/cel/transformer.go

```diff
@@ -0,0 +1,126 @@
+package transformer
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/google/cel-go/common"
+	"github.com/google/cel-go/common/ast"
+	"github.com/google/cel-go/parser"
+)
+
+func parseExpression(expression string) (*ast.AST, error) {
+	prsr, err := parser.NewParser()
+	if err != nil {
+		return nil, err
+	}
+
+	p, iss := prsr.Parse(common.NewTextSource(expression))
+	if len(iss.GetErrors()) > 0 {
+		return nil, errors.New("got errors parsing")
+	}
+
+	return p, nil
+}
+
+func TransformCounterVariable(expression string) (*string, error) {
+	var err error
+	var p *ast.AST
+	if p, err = parseExpression(expression); err != nil {
+		return nil, err
+	}
+
+	decls := make(map[string]ast.CallExpr)
+	toReplace := make([]ast.OffsetRange, 0)
+	stack := newStack()
+	stack.push(p.Expr())
+
+	for next := stack.pop(); next != nil; next = stack.pop() {
+		expr := *next
+		if expr.Kind() == ast.IdentKind && expr.AsIdent() != "descriptors" {
+			if decls[expr.AsIdent()] == nil {
```

Comment:
Should this be "simplified", and become a "[known attribute](https://github.com/Kuadrant/architecture/blob/main/rfcs/0002-well-known-attributes.md)" list that get replaced instead?
i.e. `request`, `auth`, `source`, `destination`, ... 

and for all the others, we're leaving it to the user to do "the proper" thing?

Reply:
I've done that in [this commit](https://github.com/Kuadrant/kuadrant-operator/pull/1242/commits/fd035feb05aae8183806e0239d10dd501701363b)